### PR TITLE
Clean dir name before dump

### PR DESCRIPTION
**Pull request by liornoy**
This commit changes the dump function to first invoke the "cleanDirName" function which gets the dumpSubpath, which is usually the test case name. and:
1. replace the white spaces with "_".
2. replace "/" chars with "-".

Those changes are required to make a valid directory name.

This will prevent the creation of directories named like these:
```
'should deploy the injector pod if requested'
'should deploy the operator webhook if requested'
```

### CODE REVIEW

File: reporter.go [hunk 1]
```diff
@@ -222,3 +224,9 @@ func logFileFor(dirName string, testName string, kind string) (*os.File, error)
 	}
 	return f, nil
 }
+
+func cleanDirName(dirName string) string {
+	res := strings.ReplaceAll(dirName, "/", "|")
```
Comment (ffromani) [inline reporter.go]: I'd avoide pipe in paths. If nothing else, requires more quoting. What about
```
res := strings.ReplaceAll(dirName, "/", "-")
res = strings.ReplaceAll(res, " ", "_")
```

Comment (liornoy) [reply reporter.go]: no problem I'll change that.
feel free to let me know if there are any other chars you think need to be replaced

Comment (liornoy) [reply reporter.go]: done

**Comment by fedepaol** [comment]
LGTM, thanks!

**Pull request merged**